### PR TITLE
drivers: watchdog: Convert Atmel SAM driver to DTS

### PR DESCRIPTION
**Pull request by galak**
The majority of bits where already in place, but some minor support
get the driver name from DTS was needed.  Now we select HAS_DTS_WDT
for the driver.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

**Comment by aurel32** [comment]
recheck

**Review by aurel32 — APPROVED** [review]
That looks all good to me.

File: soc/arm/atmel_sam/sam3x/dts_fixup.h [hunk 1]
```diff
@@ -39,5 +39,6 @@
 #define DT_WDT_SAM_IRQ_PRIORITY DT_ATMEL_SAM_WATCHDOG_400E1A50_IRQ_0_PRIORITY
 #define DT_WDT_SAM_LABEL        DT_ATMEL_SAM_WATCHDOG_400E1A50_LABEL
 #define DT_WDT_SAM_BASE_ADDRESS DT_ATMEL_SAM_WATCHDOG_400E1A50_BASE_ADDRESS
+#define CONFIG_WDT_0_NAME       DT_WDT_SAM_LABEL
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam/sam3x/dts_fixup.h]: Need to update prefix from `CONFIG_` to `DT_`

File: soc/arm/atmel_sam/sam4s/dts_fixup.h [hunk 2]
```diff
@@ -54,4 +54,5 @@
 #define DT_WDT_SAM_IRQ_PRIORITY DT_ATMEL_SAM_WATCHDOG_400E1450_IRQ_0_PRIORITY
 #define DT_WDT_SAM_LABEL        DT_ATMEL_SAM_WATCHDOG_400E1450_LABEL
 #define DT_WDT_SAM_BASE_ADDRESS DT_ATMEL_SAM_WATCHDOG_400E1450_BASE_ADDRESS
+#define CONFIG_WDT_0_NAME       DT_WDT_SAM_LABEL
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam/sam4s/dts_fixup.h]: ditto

File: soc/arm/atmel_sam/same70/dts_fixup.h [hunk 3]
```diff
@@ -120,6 +120,7 @@
 #define DT_WDT_SAM_IRQ_PRIORITY DT_ATMEL_SAM_WATCHDOG_400E1850_IRQ_0_PRIORITY
 #define DT_WDT_SAM_LABEL        DT_ATMEL_SAM_WATCHDOG_400E1850_LABEL
 #define DT_WDT_SAM_BASE_ADDRESS DT_ATMEL_SAM_WATCHDOG_400E1850_BASE_ADDRESS
+#define CONFIG_WDT_0_NAME       DT_WDT_SAM_LABEL
```
Comment (MaureenHelm) [inline soc/arm/atmel_sam/same70/dts_fixup.h]: ditto

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: soc/arm/atmel_sam/sam3x/dts_fixup.h (hunk 1 above)
Comment (aurel32) [reply soc/arm/atmel_sam/sam3x/dts_fixup.h]: Given the driver declares `HAS_DTS_WDT`, the `CONFIG_WDT_0_NAME` value is not filled by Kconfig and this has to be done in the fixup file. The same strategy is applied for the nRF, QMSI or MCUX drivers.

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
recheck

**Pull request merged**